karate-core/src/main/java/com/intuit/karate/filter/TagFilter.java

```diff
@@ -0,0 +1,11 @@
+package com.intuit.karate.filter;
+
+import cucumber.runtime.model.CucumberFeature;
+import cucumber.runtime.model.CucumberTagStatement;
+
+@FunctionalInterface
+public interface TagFilter {
+
+    boolean filter(CucumberFeature feature, CucumberTagStatement cucumberTagStatement) throws TagFilterException;
```

Comment:
looks nice. can you add some javadoc here as to how this works and what end-users need to do to get this working. do you think I should switch the `karate-http.properties` into this approach !